c_cpp_guide.md

```diff
@@ -0,0 +1,117 @@
+
+## Basic Step-by-Step Guide
+
+First build the latest version of LLVM.
+
+Aside: If you wish to use C/C++ standard libraries, then build the version of LLVM in the Advanced section below. That version can also be used here.
+
+```sh 
+# checkout LLVM, clang, and lld
+svn co http://llvm.org/svn/llvm-project/llvm/trunk llvm
+cd llvm/tools
+svn co http://llvm.org/svn/llvm-project/cfe/trunk clang
+svn co http://llvm.org/svn/llvm-project/lld/trunk lld
+cd ../..
+
+# build LLVM, clang, and lld
+mkdir llvm-build
+cd llvm-build
+# note: if you want other targets than WebAssembly, then delete -DLLVM_TARGETS_TO_BUILD=
+cmake -G "Unix Makefiles" -DLLVM_TARGETS_TO_BUILD= -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly ../llvm                 
+make -j 8       # WARNING, THIS CAN TAKE HOURS, NEEDS LOTS OF DISK AND RAM, AND MAY CAUSE YOUR COMPUTER TO FREEZE. IF IT ERRORS, TRY AGAIN WITHOUT -j 8
+``` 
+
+Next download and compile a wrc20 ewasm contract written in C. Note that in `main.c`, there are many arrays in global scope -- LLVM puts global arrays in WebAssembly memory, which is allows them to be used as pointer arguments to Ethereum helper functions. Before compiling, make sure that the `Makefile` has a path to `llvm-build` above, and that `main.syms` has a list of Ewasm helper functions you are using.
+
+Aside: If you are using C++, make sure to modify the Makefile to `clang++`, use `extern "C"` around the helper function declarations.
+
+```sh
+git clone https://gist.github.com/poemm/68a7b70ec353abaeae64bf6fe95d2d52.git cwrc20
+cd cwrc20
+# edit the Makefile and main.syms as described above
+make
+```
+
+The output is `main.wasm` which needs a cleanup of imports and exports to meet Ewasm requirements. For this, we use PyWebAssembly.
+
+Aside: Alternatively, one can manually cleanup. Alternatively, one can use [wasm-chisel](https://github.com/wasmx/wasm-chisel), a program in Rust which can be installed with `cargo install chisel`. The Rust version is stricter and has more features, the Python version is just enough for our use-case, and Python is available on most machines.
+
+```
+cd ..
+git clone https://github.com/poemm/pywebassembly.git
+cd pywebassembly/examples/
+python3 ewasmify.py ../../cwrc20/main.wasm
+cd ../../cwrc20
+```
+
+Check whether the command line output of `ewasmify.py` above lists only valid Ewasm imports and exports. To troubleshoot, you may wish to also inspect `main.wasm` in its text representation, so proceed to the next step with binaryen or wabt.
+
+We can convert from the `.wasm` binary format to the `.wat` (or `.wast`) text format (these are equivalent formats and can be converted back-and-forth). This conversion can be done with Binaryen's `wasm-dis`.
```

Comment:
> `.wat` (or `.wast`) text format (these are equivalent formats and can be converted back-and-forth)

Strictly speaking, this is not true. `wast` defines a "scripted" format with certain extensions to the grammar, and my understanding is that it's a superset of the `wat` format. See https://github.com/WebAssembly/design/blob/master/TextFormat.md#text-format. We've tended to elide over this in our docs thus far but I think it might be helpful to be more explicit about it. What do you think?

Reply:
Agreed. I removed `.wast`. I think that we should start using the correct extension `.wat`.

Reply:
Cool. It's clearly not critical but we can bring it up on a call and try to get everyone on the same page.